File: frontend/src/app/dtos/event-detail-dto.ts
import { ArtistDetailDto } from './artist-detail-dto';
import { HallRowsDetailDto } from './hall-rows-detail-dto';
import { SeatDto } from './seat-dto';
import { PublicFileDto } from './public-file-dto';

export interface EventDetailDto {
  id: number;
  title: string;
  startDate: Date;
  endDate: Date;
  seatPrice: number;
  standingPrice: number;
  artist: ArtistDetailDto;
  hall: HallRowsDetailDto;
  occupiedSeats: SeatDto[];
  occupiedStandings: number;
  image: PublicFileDto;
}

File: backend/src/main/java/at/ac/tuwien/sepr/groupphase/backend/endpoint/dto/EventDetailDto.java
package at.ac.tuwien.sepr.groupphase.backend.endpoint.dto;

import at.ac.tuwien.sepr.groupphase.backend.enums.EventType;
import lombok.AllArgsConstructor;
import lombok.EqualsAndHashCode;
import lombok.Getter;
import lombok.NoArgsConstructor;
import lombok.Setter;
import lombok.ToString;

import java.time.OffsetDateTime;

@Getter
@Setter
@EqualsAndHashCode
@AllArgsConstructor
@NoArgsConstructor
@ToString
public class EventDetailDto {

    private Long id;

    private String title;

    private OffsetDateTime startDate;

    private OffsetDateTime endDate;

    private Float seatPrice;

    private Float standingPrice;

    private ArtistDetailDto artist;

    private HallRowsDetailDto hall;

    private EventType type;

    private SeatDto[] occupiedSeats;

    private Integer occupiedStandings;

    private PublicFileDto image;
}
